core_backend/app/question_answer/routers.py

```diff
@@ -53,23 +55,28 @@
     responses={400: {"model": QueryResponseError, "description": "Bad Request"}},
 )
 async def llm_response(
-    user_query: QueryBase, asession: AsyncSession = Depends(get_async_session)
+    user_query: QueryBase,
+    asession: AsyncSession = Depends(get_async_session),
+    auth_token: str = Depends(auth_bearer_token),
 ) -> QueryResponse | JSONResponse:
     """
     LLM response creates a custom response to the question using LLM chat and the
     most similar embeddings to the user query in the vector db.
     """
-    # hardcode USER1_USERNAME here. Later will change this to "get_user_by_token()"
-    user = await get_user_by_username(USER1_USERNAME, asession)
+    user = await get_user_by_token(auth_token, asession)
```

Comment:
Same comment as above. Ideally we don't pass token but the user. So we move this line to get_user_from_token function. Then the endpoint dont have to know or worry about tokens.

Reply:
Yup, perfect. Will add this to final PR once auth reads from DB.